### ISSUE TITLE
Incorrect type inference in "Array and string offset validity" inspection

### ISSUE THREAD

**Issue by Riimu** [kalessil/phpinspectionsea#968]
The inspection `PHP | Php Inspections (EA Extended) | Probable bugs | Array and string offset validity` seems to detect types incorrectly in some cases.

For example, in the following code:

```php
<?php

/**
 * @param array[] $array
 * @return array
 */
function foobar(array $array) {
    $foo = [];

    foreach ($array as $key => $value) {
        $foo[] = $array[$key + 1]['bar'];
    }

    return $foo;
}
```

It complains about the `$key + 1` index with: "Resolved index type ([array]) is incompatible with possible [string, int]". There is however, absolutely no way that operation could result as an array.

**Comment by kalessil** [comment]
Thank you for reporting @Riimu , I'll investigate.

**Comment by kalessil** [comment]
Fixed!
